pkg/startup/startup.go

```diff
@@ -250,6 +250,19 @@ func configureAndCheckIPAddressSubnets(ctx context.Context, cli client.Interface
 		terminate()
 	}
 
+	if node.Spec.BGP.IPv4Address == "" && node.Spec.BGP.IPv6Address == "" {
+		if os.Getenv("CALICO_NETWORKING_BACKEND") != "none" {
+			log.Error("No IP Addresses configured, and autodetection is not enabled")
```

Comment:
I don't think this error makes sense any more - we moved the check and it no longer knows if auto-detection is enabled or not. All it knows is that there are no addresses and that it needs them.

I think we need to reword the error if we're going to move it out here.

e.g., "No IP addresses configured or detected, but they are required for Calico networking"

Reply:
Still thinking through whether it's OK to actually move this check here.

Reply:
Sure, I have adjusted the UT with this change. Let me know if it makes sense given the usage there.

Reply:
The fields themselves aren't used if not using Calico networking. However, it's bad form to silently modify the user's intent (e.g., if a user sets one of these fields, and we allow it, we shouldn't come along and clear them out later).

One somewhat contrived scenario might be that they are planning on switching to Calico networking and this is a transient state.